DataFormats/Portable/interface/PortableHostCollection.h

```diff
@@ -47,27 +50,86 @@ class PortableHostCollection {
   PortableHostCollection& operator=(PortableHostCollection const&) = delete;
 
   // movable
-  PortableHostCollection(PortableHostCollection&&) = default;
-  PortableHostCollection& operator=(PortableHostCollection&&) = default;
+  PortableHostCollection(PortableHostCollection&& other):
+    buffer_(std::move(other.buffer_)),
+    layout_(std::move(other.layout_)),
+    view_(std::move(other.view_))
+          {
+            other.buffer_.reset();
+            other.view_ = View();
+          }
+  PortableHostCollection& operator=(PortableHostCollection&& other) {
+    // Protection for self-assignment without if-statements, following
+    // what was suggested in the C++ core guidelines
+    {
+      auto tmp = std::move(other.buffer_);
+      other.buffer_.reset();
+      buffer_.reset();
+      buffer_ = std::move(tmp);
+    }
+    {
+      auto tmp = std::move(other.view_);
+      other.view_ = View();
+      view_ = View();
+      view_ = std::move(tmp);
+    }
```

Comment:
it _should_ (to be verified) safe to do just
```suggestion
    {
      view_ = other.view_;  // self-assignment is safe
      other.view_ = View();
    }
```

Reply:
however, what about the `layout_` ?

Reply:
> it _should_ (to be verified) safe to do just

This is because `View` contains only pointers and fundamental types? Right, then I'd also expect the self-assignment to be safe.

> however, what about the `layout_` ?

"Whoops", I guess, thanks for catching. Then I need to think a test that would demonstrate it failing.